util/shortHumanReadableDate.ts

```diff
@@ -0,0 +1,9 @@
+export function shortHumanReadableDate(date: Date, locale?: string) {
```

Comment:
I would actually prefer a generic DateTimeFormat function that accepts the props of DateTimeFormat.

Reply:
> The way I see it, this diminishes the intended value of this shortHumanReadableDate, as it was meant as an abstraction.

I don't see the value of having a helper used in one (see the rule of 3 https://blog.codinghorror.com/rule-of-three/) place for formatting dates, you instead want to simply invoke Intl.DateTimeFormat directly from your Component, then.

Also remember that you shouldn't instantiate `Intl.DateTimeFormat` within a component. Please simply crate the format you want on the root of your Component file and then invoke the `myTimeFormat.format(myDateObj)` within the component. Remember to memoize it if you don't want a new date/format to happen on every rerender.

> By updating it and leaving it as broad as possible, people would need to check the Intl.DateTimeFormat's options parameter API, and understand how to achieve what they want, so its usage would be a bit less straightforward, and not so consistent, as some might configure them a bit differently:

Exactly, that's the conclusion I wanted you to reach. It's a useless abstraction.

Reply:
Thank you. Great read.

I also agree in avoiding early abstractions but had incorrectly assumed that this util function would be used in several places.

I did not understand what have mentioned about not instantiating Intl.DateTimeFormat within a component. Could you guide me on what you meant by that?

Reply:
Right, so Intl.DateTimeFormat constructor returns an instance of DateTimeFormat that you can use against any date object. It is something static. You don't need to call the constructor inside a React Component as it will never change, meaning:

```tsx
const shortDateTimeFormat = new Intl.DateTimeFormat({ ... });

const MyComponent = ({ date }) => {
  const formattedDate = useMemo(() => 
    shortDateTimeFormat.format(date), [date]);

  ...
}
```